src/Jobs/Middleware/RecordTransaction.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace AG\ElasticApmLaravel\Jobs\Middleware;
+
+use AG\ElasticApmLaravel\Agent;
+use AG\ElasticApmLaravel\Collectors\JobCollector;
+
+class RecordTransaction
+{
+    /**
+     * Wrap the job processing in an APM transaction.
+     *
+     * @param mixed    $job
+     * @param callable $next
+     *
+     * @return mixed
+     */
+    public function handle($job, $next)
+    {
+        if (false === config('elastic-apm-laravel.active') || false === config('elastic-apm-laravel.cli.active')) {
+            return $next($job);
+        }
+
+        /** @var Agent */
+        $agent = app(Agent::class);
+        /** @var JobCollector */
+        $collector = $agent->getCollector(JobCollector::getName());
```

Comment:
@arkaitzgarro I'm not sure what to do here. My issue is that I can no longer call `getName` because it's not static. I see 2 obvious solutions

- I believe I saw @countless-integers mention using a constant instead? That would help here.
- Maybe I should just use the facade instead. I thought it was nice to keep these measurements within the job but maybe there's no reason they need to be. I think I'll go that route but I'll leave this comment anyway in case you have other opinions.

Reply:
I'm also thinking that we could ditch the `getName` method, and use the class of the collector as a name. That could also prevent any problem if a custom collector uses the same name as we do with default collectors.